Src/ElasticScale.Client/ShardManagement/SchemaInformation/SchemaInfo.cs

```diff
@@ -55,22 +71,35 @@ public ReadOnlyCollection<ShardedTableInfo> ShardedTables
         /// </summary>
         public ReadOnlyCollection<ReferenceTableInfo> ReferenceTables
         {
-            get { return _referenceTableSet.ToList().AsReadOnly(); }
+            get { return _referenceTables.ToList().AsReadOnly(); }
         }
 
         /// <summary>
         /// Initialize any non-DataMember objects post deserialization.
         /// </summary>
         [OnDeserialized()]
-        private void SetValuesOnDeserializing(StreamingContext context)
+        private void SetValuesOnDeserialized(StreamingContext context)
         {
             Initialize();
         }
 
+        /// <summary>
+        /// Initializes this instance after construction or deserialization. 
+        
+        /// </summary>
         private void Initialize()
         {
-            _shardedTableSet = _shardedTableSet ?? new HashSet<ShardedTableInfo>();
-            _referenceTableSet = _referenceTableSet ?? new HashSet<ReferenceTableInfo>();
+            // If _shardedTables is null after deserialization, then set it to _shardedTablesWrongName
+            // instead (in case we deserialized the v1.1.0 format). If that is also null, then just set 
+            // it to an empty HashSet. 
+            _shardedTables = _shardedTables ?? _shardedTablesAlternateName ?? new HashSet<ShardedTableInfo>();
+            // Null out _shardedTablesWrongName so that we don't serialize it back
```

Comment:
nit: _shardedTablesAlternateName instead of _shardedTablesWrongName in this and above comment.
All other changes look good to me.